lib/redis-client.js

```diff
@@ -0,0 +1,95 @@
+var RSVP  = require('rsvp');
+var redis = require('then-redis');
+var SilentError = require('ember-cli/lib/errors/silent');
+var fs = require('fs');
+var tunnel = require('tunnel-ssh');
+var untildify = require('untildify');
+
+function error(message) {
+  return RSVP.Promise.reject(new SilentError(message));
+}
+
+function wrap(methodName) {
+  return function () {
+    var args = Array.prototype.slice.call(arguments);
+    return this._connection.then(function (client) {
+      return client[methodName].apply(client, args);
+    });
+  };
+}
+
+function promisifyNodeCallback(target, method) {
```

Comment:
We are already using RSVP here anyway. There's [RSVP.denodify](http://emberjs.com/api/classes/RSVP.html#method_denodeify) to do the same thing.